frontend/public/components/pdb/PodDisruptionBudgetField.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { useTranslation } from 'react-i18next';
+import { PodDisruptionBudgetKind, LabelSelector } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '../utils/k8s-watch-hook';
+import { ResourceLink, DetailsItem } from '../utils';
+import { PodDisruptionBudgetModel } from '../../models';
+
+const AvaiallabilityRequirement: React.FC<AvaiallabilityRequirementProps> = ({ pdb, replicas }) => {
+  const { t } = useTranslation();
+  return (
+    <>
+      {pdb?.spec?.minAvailable
+        ? t('public~Min Available {{minAvailable}} of {{count}} pod', {
+            minAvailable: pdb.spec.minAvailable ?? 1,
+            count: replicas,
+          })
+        : t('public~Max Unavailable {{maxUnavailable}} of {{count}} pod', {
+            maxUnavailable: pdb?.spec?.maxUnavailable ?? 1,
+            count: replicas,
+          })}
+    </>
+  );
+};
+
+export const PodDisruptionBudgetField: React.FC<PodDisruptionBudgetFieldProps> = ({ obj }) => {
+  const { t } = useTranslation();
+  const [resources] = useK8sWatchResource<PodDisruptionBudgetKind[]>({
+    kind: PodDisruptionBudgetModel.kind,
+    isList: true,
+    namespaced: true,
+    namespace: obj.metadata.namespace,
+  });
+  const pdb = resources.find((f) =>
+    new LabelSelector(f.spec.selector.matchLabels).matchesLabels(
```

Comment:
Need to handle `matchExpressions`. You might want to pull this out into a utility somewhere since this code is repeated in several places.